dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/validation/validator/event/DateValidatorTest.java

```diff
@@ -246,12 +246,48 @@ void testEventIsNotValidWhenDateBelongsToExpiredPeriod() {
     assertHasError(reporter, event, E1047);
   }
 
-  private Program getProgramWithRegistration() {
+  @Test
+  void shouldPassValidationForEventWhenDateBelongsToPastPeriodWithZeroExpiryDays() {
+    // given
+    when(preheat.getProgram(MetadataIdentifier.ofUid(PROGRAM_WITH_REGISTRATION_ID)))
+        .thenReturn(getProgramWithRegistration(0));
+    Event event = new Event();
+    event.setEvent(UID.generate());
+    event.setProgram(MetadataIdentifier.ofUid(PROGRAM_WITH_REGISTRATION_ID));
+    event.setOccurredAt(sevenDaysAgo());
+    event.setStatus(EventStatus.ACTIVE);
+
+    // when
+    validator.validate(reporter, bundle, event);
+
+    // then
+    assertIsEmpty(reporter.getErrors());
+  }
+
+  @Test
+  void shouldFailValidationForEventWhenScheduledDateBelongsToFuturePeriod() {
```

Comment:
Either the name of the test or the assertion is wrong here.
AFAIK the name seems correct but we should get an error than in the reporter.

Reply:
I think the name was incorrect. ScheduledDate should be allowed for future periods. I have changed the name to reflect that is should pass.